### PR TITLE
Feat(medusa-js, medusa-react): Upload endpoints in medusa js and react

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- Add `delete` and `getPresignedDownloadUrl` to `medusa-js` and `medusa-react`
- remove `upload` from medusa-js and medusa-react

**How**
- following existing patterns for medusa-js and medusa-react 

**Why**
- to support file operations in client libraries
- upload was not working in medusa-js and it would be better to remove it for now than have a method that doesn't work.

__note__: I am currently working towards getting upload in medusa-js to work, but it could be for another pr

### CODE REVIEW

File: packages/medusa-react/src/hooks/admin/uploads/mutations.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  AdminDeleteUploadReq,
+  AdminCreateUploadsFileDownloadUrlReq,
+  AdminCreateUploadsFileDownloadUrlRes,
+} from "@medusajs/medusa"
+import { Response } from "@medusajs/medusa-js"
+import { useMutation, UseMutationOptions, useQueryClient } from "react-query"
+import { useMedusa } from "../../../contexts"
+import { buildOptions } from "../../utils/buildOptions"
+
+// export const useAdminUploadFile = (
```
Comment (adrien2p) [inline packages/medusa-react/src/hooks/admin/uploads/mutations.ts:12]: question: Did all the file creation have been deleted?

**Review by adrien2p — COMMENTED** [review]
LGTM 🎉 few dew comment only

File: packages/medusa/src/api/routes/admin/uploads/get-download-url.ts [hunk 2]
```diff
@@ -26,18 +26,17 @@ import { validator } from "../../../../utils/validator"
...
-    fileKey: validated.file_key,
+    fileKey: (req.validatedBody as AdminCreateUploadsFileDownloadUrlReq)
+      .file_key,
   })
 
   res.status(200).send({ download_url: url })
 }
 
-class AdminGetUploadsFileDownloadUrlReq {
+export class AdminCreateUploadsFileDownloadUrlReq {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/uploads/get-download-url.ts:39]: todo: Follow payload naming convention
```suggestion
export class AdminPostUploadsDownloadUrlReq {
```

File: packages/medusa/src/api/routes/admin/uploads/index.ts [hunk 3]
```diff
@@ -16,22 +18,31 @@ export default (app) => {
...
 }
 
 export type AdminUploadRes = {
-  uploads: unknown[]
+  uploads: { url: string }[]
 }
 
 export type AdminDeleteUploadRes = DeleteResponse
 
+export type AdminCreateUploadsFileDownloadUrlRes = {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/uploads/index.ts:42]:
```suggestion
export type AdminUploadsDownloadUrlRes = {
```

File: packages/medusa/src/api/routes/admin/uploads/delete-upload.ts [hunk 4]
```diff
@@ -27,7 +24,7 @@ export default async (req, res) => {
     .send({ id: validated.file_key, object: "file", deleted: true })
 }
 
-class AdminDeleteUploadReq {
+export class AdminDeleteUploadReq {
```
Comment (olivermrbl) [inline packages/medusa/src/api/routes/admin/uploads/delete-upload.ts:27]:
```suggestion
export class AdminDeleteUploadsReq {
```

**Review by olivermrbl — COMMENTED** [review]
Overall LGTM, just added a couple of comments on naming conventions 💪 

Note, I've just added the convention in [our document](https://www.notion.so/medusajs/Medusa-Engineering-Conventions-dcdeccf9eca14ea1975b19b92a44c94d#18732732166741c79e943ce48600ca50) as part of reviewing this PR, so you could not have known 😅

**Review by olivermrbl — APPROVED** [review]
LGTM! 🙌

**Pull request merged**